node/src/store.rs

```diff
@@ -1,14 +1,361 @@
+use async_trait::async_trait;
+use celestia_types::ExtendedHeader;
+use core::fmt::Debug;
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use tendermint::Hash;
+use thiserror::Error;
+use tracing::{error, info, instrument};
+
+type Result<T, E = StoreError> = std::result::Result<T, E>;
+
+pub type WrappedStore = Arc<Box<dyn Store + Sync + Send>>;
+
+#[async_trait]
+pub trait Store: Send + Sync + Debug {
```

Comment:
If you have `Send + Sync` in the boxed type, then it is not needed to be bound also here. Keep one of the two.

I'm not sure what I prefer.

Reply:
Usually the `Send + Sync` is not bound to the trait itself, unless if it is a hard requirement.